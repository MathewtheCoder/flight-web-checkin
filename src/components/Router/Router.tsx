import React from "react";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import routes from "routes";

type RouteInfo = {
  component: () => JSX.Element;
  path: string;
};
const Router = (): JSX.Element => (
  <BrowserRouter>
    <Switch>
      {routes.map((route: RouteInfo) => (
        <Route
          path={route.path}
          component={route.component}
          key={route.path}
          exact
        />
      ))}
    </Switch>
  </BrowserRouter>
);

export default Router;
