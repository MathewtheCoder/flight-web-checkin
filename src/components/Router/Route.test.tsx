import React from 'react';
import { shallow, toJson } from 'tests/enzyme';
import Router from './Router';

describe('Router tests', () => {

  it('renders correctly Router', () => {
    const wrapper = shallow(<Router />);
    // Expect the wrapper object to be defined
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});