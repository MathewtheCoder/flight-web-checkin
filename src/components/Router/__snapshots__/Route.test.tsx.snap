// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router tests renders Router component 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="/"
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/personal-info"
      path="/personal-info"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/review-info"
      path="/review-info"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/checkin-success"
      path="/checkin-success"
    />
  </Switch>
</BrowserRouter>
`;

exports[`Router tests renders correctly Router 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="/"
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/personal-info"
      path="/personal-info"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/review-info"
      path="/review-info"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/checkin-success"
      path="/checkin-success"
    />
  </Switch>
</BrowserRouter>
`;

exports[`Router tests renders list-items 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="/"
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/personal-info"
      path="/personal-info"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/review-info"
      path="/review-info"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/checkin-success"
      path="/checkin-success"
    />
  </Switch>
</BrowserRouter>
`;
